src/features/goblins/bank/components/WithdrawTokens.tsx

```diff
@@ -40,18 +46,31 @@ export const WithdrawTokens: React.FC<Props> = ({ onWithdraw }) => {
     setIsLoading(true);
 
     const load = async () => {
-      const { game: state } = await getOnChainState({
-        id: goblinState.context.state.id as number,
-        farmAddress: goblinState.context.state.farmAddress as string,
+      const { game: onChainState } = await getOnChainState({
+        id: state.id as number,
+        farmAddress: state.farmAddress as string,
       });
 
-      setBalance(state.balance);
+      setBalance(onChainState.balance);
       setIsLoading(false);
     };
 
     load();
   }, []);
 
+  useEffect(() => {
+    // Use base 1000
+    let _tax =
+      getTax(typeof amount !== "string" ? amount : new Decimal(0)) / 10;
```

Comment:
Nice idea! 🏆 

Since this is business logic, I would suggest moving the logic  `if (state.inventory["Liquidity Provider"]?.gte(1)) {
      _tax = _tax / 2;
    }` into the `getTax` functionality.

This will help keep this component UI focussed

Reply:
done

with skill
![sfl-withdraw-lp-badge-resolved-with-skill-2](https://user-images.githubusercontent.com/89294757/169748803-1bfcf24e-8f72-473d-ac66-8afdc954fb41.PNG)
![sfl-withdraw-lp-badge-resolved-with-skill](https://user-images.githubusercontent.com/89294757/169748812-f153aded-9359-44a2-9887-4643be4bafed.PNG)

no skill
![sfl-withdraw-lp-badge-resolved-no-skill](https://user-images.githubusercontent.com/89294757/169748815-9cb6f2f7-9af2-4a6b-8333-e741dbfd0327.PNG)